services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -50,15 +50,15 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     responseBody: UpdateFiledResponse
   )(
     implicit ec: ExecutionContext
-  ): AsyncResult[UpdateFiledResponse] =
+  ): AsyncResult[(UpdateFiledResponse, PipelineVersion)] =
     createTag(repositoryId, version).map {
-      case Right(_)        => Right(responseBody)
+      case Right(_)        => Right(responseBody, version)
       case Left(exception) => Left(exception)
     }
 
   override def updateFile(project: Project, projectFile: ProjectFile, userVersion: Option[PipelineVersion])(
     implicit ec: ExecutionContext
-  ): AsyncResult[UpdateFiledResponse] = {
+  ): AsyncResult[(UpdateFiledResponse, PipelineVersion)] = {
```

Comment:
It is ok to sometimes use tuples internally. But you should avoid using tuples in public methods. Let's create a new case class that will model your response. It will resolve two problems
1) Tuples are hard two read. What is `PipelineVersion` in this case? Is it new version? Is it previous version? Is it current version?
2) Tuples are not scaling well. If we add a new field to a returned value, most likely we will have to go through a lot of usages of such method and modify them, because return type would be different.
Once you add this class, don't forget to change the return type of `handleCreateTag` as well.